lang/cs/Org.Apache.REEF.IMRU.Tests/TestTaskManager.cs

```diff
@@ -0,0 +1,702 @@
+            var masterTask = CreateMockRunningTask(MasterTaskId);
+            taskManager.RecordRunningTaskDuringSystemFailure(masterTask, TaskManager.CloseTaskByDriver);
+            Assert.Equal(TaskState.TaskWaitingForClose, taskManager.TaskState(MasterTaskId));
+
+            //// The task 2 could be failed by communication before receiving close event
+            var failedTask2 = CreateMockFailedTask(MapperTaskIdPrefix + 2, TaskManager.TaskGroupCommunicationError);
+            taskManager.RecordFailedTaskDuringSystemShuttingDownState(failedTask2);
+
+            //// master task failed because receiving close event
+            var failedMasterTask = CreateMockFailedTask(MasterTaskId, TaskManager.TaskKilledByDriver);
+            taskManager.RecordFailedTaskDuringSystemShuttingDownState(failedMasterTask);
+
+            Assert.True(taskManager.AllInFinalState());
+        }
+
+        /// <summary>
+        /// Tests a non running task fails before all the tasks are running
+        /// </summary>
+        [Fact]
+        public void TestFailedNoRunningTaskBeforeAllTasksAreRunningScenario()
+        {
+            var taskManager = TaskManagerWithTasksAdded();
+            taskManager.SubmitTasks();
+
+            taskManager.RecordRunningTask(CreateMockRunningTask(MapperTaskIdPrefix + 1));
+
+            //// Evaluator error
+            var failedTask2 = CreateMockFailedTask(MapperTaskIdPrefix + 2, TaskManager.TaskSystemError);
+            taskManager.RecordFailedTaskDuringRunningOrSubmissionState(failedTask2);
+            Assert.Equal(TaskState.TaskFailedBySystemError, taskManager.TaskState(MapperTaskIdPrefix + 2));
+
+            //// there is no any running task yet
+            taskManager.CloseAllRunningTasks(TaskManager.CloseTaskByDriver);
+
+            //// master task is running, close it
+            var masterTask = CreateMockRunningTask(MasterTaskId);
+            taskManager.RecordRunningTaskDuringSystemFailure(masterTask, TaskManager.CloseTaskByDriver);
+            Assert.Equal(TaskState.TaskWaitingForClose, taskManager.TaskState(MasterTaskId));
+
+            //// The task 1 could be failed by communication before receiving close event
+            var failedTask1 = CreateMockFailedTask(MapperTaskIdPrefix + 1, TaskManager.TaskGroupCommunicationError);
+            taskManager.RecordFailedTaskDuringSystemShuttingDownState(failedTask1);
+
+            //// master task failed could be failed by communication error as well
+            var failedMasterTask = CreateMockFailedTask(MasterTaskId, TaskManager.TaskGroupCommunicationError);
+            taskManager.RecordFailedTaskDuringSystemShuttingDownState(failedMasterTask);
+
+            Assert.True(taskManager.AllInFinalState());
+        }
+
+        /// <summary>
+        /// Creates a TaskManger for testing with defined constants in the class
+        /// </summary>
+        /// <returns></returns>
+        private static TaskManager CreateTaskManager()
```

Comment:
Please delete this method and add default values to `CreateTaskManager(int numTasks, string masterTaskId)` instead